CONTRIBUTING.md

```diff
@@ -49,29 +119,138 @@ given file in the **Files** view.
+particular outcome with Markdown, Bootstrap, JQuery, or something else, have
+a look at `test.md` (which renders in the site at `/test/`).
+
+### Creating tabs
+
+The use of tabs, as on pages like [https://docs.docker.com/engine/api/](/engine/api/), requires the use of HTML. The tabs use Bootstrap CSS/JS, so refer to those docs for more advanced usage. For a basic horizontal tab set, copy/paste starting from this code and implement from there. Keep an eye on those `href="#id"` and `id="id"` references as you rename, add, and remove tabs.
+
+```html
+<ul class="nav nav-tabs">
+  <li class="active"><a data-toggle="tab" data-target="#tab1">TAB 1 HEADER</a></li>
+  <li><a data-toggle="tab" data-target="#tab2">TAB 2 HEADER</a></li>
+</ul>
+<div class="tab-content">
+  <div id="tab1" class="tab-pane fade in active">TAB 1 CONTENT</div>
+  <div id="tab2" class="tab-pane fade">TAB 2 CONTENT</div>
+</div>
+```
+
+For more info and a few more permutations, see [test.md](/test.md).
+
+### Running in-page Javascript
+
+If you need to run custom Javascript within a page, and it depends upon JQuery
+or Bootstrap, make sure the `<script>` tags are at the very end of the page,
+after all the content. Otherwise, the script may try to run before JQuery and
+Bootstrap JS are loaded.
+
+### Images
+
+Don't forget to remove images that are no longer used. Keep the images sorted
+in the local `images/` directory, with names that naturally group related images
+together in alphabetical order. For example,  use `settings-file-share.png`
+and `settings-proxies.png` instead of `file-share-settings.png` and
+`proxies-settings.png`.
+
+You may also use numbers, especially in the case of a
+sequence, for example, `run-only-the-images-you-trust-1.svg`,
+`run-only-the-images-you-trust-2.png`, `run-only-the-images-you-trust-3.png`.
+When applicable, capture windows rather than the rectangular regions. This
+eliminates unpleasant background and saves the editors the need to crop images.
+
+On Mac, capture windows without shadows. To do this, when  you press
+<kbd>Command-Shift-4</kbd>, press <kbd>Option</kbd> while clicking on the window. To disable shadows completely, run:
+
+```bash
+$ defaults write com.apple.screencapture disable-shadow -bool TRUE
+$ killall SystemUIServer  # restart it.
+```
+
+You can restore the shadows later using `-bool FALSE`.
+
+To keep the Git repository light, _please_ compress the images
+(losslessly). On Mac, you can use [ImageOptim](https://imageoptim.com) for
+instance. Be sure to compress the images **before** adding them to the
+repository. Compressing images after adding them to the repository actually worsens the impact on the Git repo (however, ut still optimizes the bandwidth during browsing).
+
+### Style guide
+
+Docker does not currently maintain a style guide. Follow the examples set by the existing documentation and use your best judgment.
+
+## Build and preview the docs locally
```

Comment:
Wondering if this would perhaps still be useful in the README (explaining users how to view the docs locally, even if they're not interested in contributing)